### PR TITLE
End to end ci

### PR DESCRIPTION
**Pull request by k-rister**

### CODE REVIEW

**Comment by k-rister** [comment]
@rafaelfolco @atheurer
This is going to need heavy scrutiny and almost certainly cleanup.  There are also some temporary changes in here that will have to be pulled out before final review and merge.

This PR is also dependent on https://github.com/perftool-incubator/rickshaw/pull/184

File: crucible-install.sh [hunk 1]
```diff
@@ -13,7 +13,14 @@ if [ "${PWD}" != "${SCRIPT_DIR}" ]; then
     fi
 fi
 git_status=$(git status --porcelain=2 --untracked-files=no --branch 2>&1)
+git_use_default=0
 if echo -e "${git_status}" | grep -q "not a git repository"; then
+    git_use_default=1
+fi
+if ! echo -e "${git_status}" | grep "branch\.upstream"; then
+    git_use_default=1
```
Comment (rafaelfolco) [inline crucible-install.sh]: hacky

File: crucible-install.sh [hunk 2]
```diff
@@ -58,6 +31,49 @@ EC_FAIL_AUTH_SET=13
...
+    if [ "${PWD}" != "${SCRIPT_DIR}" ]; then
+        if ! pushd ${SCRIPT_DIR} > /dev/null; then
+            echo "WARNING: Failed to pushd to ${SCRIPT_DIR}"
+            git_use_default=1
+        fi
+    fi
+
+    git_status=$(git status --porcelain=2 --untracked-files=no --branch 2>&1)
+    if echo -e "${git_status}" | grep -q "not a git repository"; then
+        git_use_default=1
```
Comment (rafaelfolco) [inline crucible-install.sh:57]: standalone script downloaded --> install from origin/master

File: crucible-install.sh [hunk 3]
```diff
@@ -58,6 +31,49 @@ EC_FAIL_AUTH_SET=13
...
+    fi
+    if ! echo -e "${git_status}" | grep "branch\.upstream"; then
+        git_use_default=1
+    fi
```
Comment (rafaelfolco) [inline crucible-install.sh:61]: upstream is not fetched in these cases:
1) not a fork, user cloned with ssh the main repo --> install from origin/master
2) PR --> enforce --git-repo (or change how the pr checkout works)

File: crucible-install.sh [hunk 4]
```diff
@@ -58,6 +31,49 @@ EC_FAIL_AUTH_SET=13
...
+
+    if [ "${git_use_default}" == 0 ]; then
+        git_tracking=$(echo "${git_status}" | grep "branch\.upstream" | awk '{ print $3 }')
+        git_remote_branch=$(echo "${git_tracking}" | awk -F'/' '{ print $2 }')
+        git_remote_name=$(echo "${git_tracking}" | awk -F'/' '{ print $1 }')
+        git_remote_url=$(git remote get-url ${git_remote_name})
+
+        GIT_REPO="${git_remote_url}"
+        GIT_BRANCH="${git_remote_branch}"
```
Comment (rafaelfolco) [inline crucible-install.sh:70]: auto-detection case
you need to fork the main repo to test changes, there is no way to edit a file on the repo cloned with ssh and test it

**Review by rafaelfolco — APPROVED** [review]
lgtm
this looks pretty good !!
perhaps --git-repo opens too many possibilities for problems...
reminder: update INSTALL.md to reflect the new nstaller args...

File: crucible-install.sh [hunk 5]
```diff
@@ -2,12 +2,35 @@
...
+    GIT_REPO="https://github.com/perftool-incubator/crucible.git"
+    GIT_BRANCH="master"
+else
+    git_tracking=$(echo "${git_status}" | grep "branch\.upstream" | awk '{ print $3 }')
+    git_remote_branch=$(echo "${git_tracking}" | awk -F'/' '{ print $2 }')
+    git_remote_name=$(echo "${git_tracking}" | awk -F'/' '{ print $1 }')
+    git_remote_url=$(git remote get-url ${git_remote_name})
+
+    GIT_REPO="${git_remote_url}"
+    GIT_BRANCH="${git_remote_branch}"
```
[the remark this answers is not in the document]
Comment (k-rister) [reply crucible-install.sh]: This gets some history, but it doesn't change anything else about the state of the repository:

https://github.com/perftool-incubator/crucible/runs/3247623465?check_suite_focus=true
https://github.com/k-rister/crucible/runs/3247623305?check_suite_focus=true

File: crucible-install.sh [hunk 6]
```diff
@@ -34,14 +39,17 @@ function usage {
 
     Crucible installer script.
 
-    Usage: $0 --registry <value> --auth-file <value> [ opt ]
+    Usage: $0 [--client-server-registry <value> --client-server-auth-file <value> | --no-client-server-registry ] [ opt ]
```
[the remark this answers is not in the document]
Comment (k-rister) [reply crucible-install.sh]: I need to do this...

File: crucible-install.sh (hunk 1 above)
Comment (k-rister) [reply crucible-install.sh]: Why is this "hacky"?

File: crucible-install.sh (hunk 2 above)
Comment (k-rister) [reply crucible-install.sh:57]: I should probably add comments to note what the different use cases are that will lead to these different conclusions.

File: crucible-install.sh (hunk 6 above)
[the remark this answers is not in the document]
Comment (k-rister) [reply crucible-install.sh]: this is done

File: crucible-install.sh (hunk 2 above)
Comment (k-rister) [reply crucible-install.sh]: done

**Review by rafaelfolco — APPROVED** [review]
lgtm

**Pull request merged**